src/box/lua/config/applier/console.lua

```diff
@@ -10,34 +14,49 @@ local function socket_file_to_listen_uri(file)
     return ('unix/:./%s'):format(file)
 end
 
+local function close_listening_socket()
+    if listening_socket ~= nil then
+        listening_socket:close()
+        listening_socket = nil
+        listening_socket_file = nil
+    end
+end
+
 local function apply(config)
     local configdata = config._configdata
     local enabled = configdata:get('console.enabled', {use_default = true})
     if not enabled then
         log.debug('console.apply: console is disabled by the ' ..
             'console.enabled option; skipping...')
+        -- If the console is disabled, we must close the active socket.
+        close_listening_socket()
         return
     end
 
-    local socket_file = configdata:get('console.socket', {use_default = true})
-    assert(socket_file ~= nil)
+    local new_socket_file = configdata:get(
+        'console.socket', {use_default = true}
+    )
+    assert(new_socket_file ~= nil)
 
     -- The same socket file is pointed by different paths before
     -- and after first box.cfg() if the `process.work_dir` option
     -- is set.
     local iconfig_def = configdata._iconfig_def
-    socket_file = instance_config:prepare_file_path(iconfig_def, socket_file)
-
-    local listen_uri = socket_file_to_listen_uri(socket_file)
-    log.debug('console.apply: %s', listen_uri)
-
-    -- Ignore 'Address already in use' error, because it is
-    -- natural effect of reloading the configuration. All others
-    -- errors are re-raised and lead to failure of apply of the
-    -- configuration.
-    local ok, res = pcall(console.listen, listen_uri)
-    if not ok and errno() ~= errno.EADDRINUSE then
-        error(res)
+    new_socket_file = instance_config:prepare_file_path(
+        iconfig_def, new_socket_file)
+    local new_socket_file_path = fio.abspath(new_socket_file)
+
+    -- If console.socket has changed we must close the listening socket.
+    if listening_socket_file ~= new_socket_file_path and
```

Comment:
The naming is inconsistent. One `<...>_file` valiable holds a relative path, another holds an absolute path. I propose to use `_file` for a relative path and `_abspath` for the absolute one.

<details>
<summary>I proopse the following changes.</summary>

```diff
diff --git a/src/box/lua/config/applier/console.lua b/src/box/lua/config/applier/console.lua
index 7de5ab8f3a..c34d409280 100644
--- a/src/box/lua/config/applier/console.lua
+++ b/src/box/lua/config/applier/console.lua
@@ -3,9 +3,9 @@ local log = require('internal.config.utils.log')
 local instance_config = require('internal.config.instance_config')
 local fio = require('fio')
 
--- active socket and its location
+-- Active socket and its location.
 local listening_socket = nil
-local listening_socket_file = nil
+local listening_socket_abspath = nil
 
 local function socket_file_to_listen_uri(file)
     if file:startswith('/') or file:startswith('./') then
@@ -18,7 +18,7 @@ local function close_listening_socket()
     if listening_socket ~= nil then
         listening_socket:close()
         listening_socket = nil
-        listening_socket_file = nil
+        listening_socket_abspath = nil
     end
 end
 
@@ -33,30 +33,28 @@ local function apply(config)
         return
     end
 
-    local new_socket_file = configdata:get(
-        'console.socket', {use_default = true}
-    )
+    local new_socket_file = configdata:get('console.socket',
+        {use_default = true})
     assert(new_socket_file ~= nil)
 
     -- The same socket file is pointed by different paths before
     -- and after first box.cfg() if the `process.work_dir` option
     -- is set.
     local iconfig_def = configdata._iconfig_def
-    new_socket_file = instance_config:prepare_file_path(
-        iconfig_def, new_socket_file)
-    local new_socket_file_path = fio.abspath(new_socket_file)
+    new_socket_file = instance_config:prepare_file_path(iconfig_def,
+        new_socket_file)
+    local new_socket_abspath = fio.abspath(new_socket_file)
 
     -- If console.socket has changed we must close the listening socket.
-    if listening_socket_file ~= new_socket_file_path and
-            listening_socket ~= nil then
+    if listening_socket_abspath ~= new_socket_abspath then
         close_listening_socket()
     end
 
-    if listening_socket_file == nil then
+    if listening_socket_abspath == nil then
         local listen_uri = socket_file_to_listen_uri(new_socket_file)
         log.debug('console.apply: %s', listen_uri)
         listening_socket = console.listen(listen_uri)
-        listening_socket_file = new_socket_file_path
+        listening_socket_abspath = new_socket_abspath
     end
 end
 
```
</details>

Reply:
Corrected